mlflow/utils/autologging_utils/__init__.py

```diff
@@ -429,3 +429,87 @@ def autologging_is_disabled(integration_name):
         return get_autologging_config(integration_name, "disable_for_unsupported_versions", False)
 
     return False
+
+
+def _get_training_session():
+    """
+    Returns a session manager for nested autologging runs.
+    """
+    # NOTE: The current implementation doesn't guarantee thread-safety, but that's okay for now
+    # because:
+    # 1. We don't currently have any use cases for allow_children=True.
+    # 2. The list append & pop operations are thread-safe, so we will always clear the session stack
+    #    once all _SklearnTrainingSessions exit.
+    class _TrainingSession(object):
+        _session_stack = []
+
+        def __init__(self, clazz, allow_children=True):
+            """
+            A session manager for nested autologging runs.
+
+            :param clazz: A class object that this session originates from.
+            :param allow_children: If True, allows autologging in child sessions.
+                                   If False, disallows autologging in all descendant sessions.
+
+            Example:
+
+            >>> class Parent: pass
+            >>> class Child: pass
+            >>> class Grandchild: pass
+
```

Comment:
the doc code missing a line ?
```
_SklearnSession = _get_training_session()
```

seems like the example code should be put in `_get_training_session` doc

Reply:
Good point. Move the example in the `_get_training_session` doc